projects/storefrontlib/src/shared/services/trigger/config/trigger.service.ts

```diff
@@ -1,28 +1,61 @@
-import { Injectable, ComponentFactory } from '@angular/core';
+import {
+  Injectable,
+  ComponentFactory,
+  ComponentFactoryResolver,
+} from '@angular/core';
 import {
   TriggerOutletMapping,
   TriggerInlineMapping,
   TriggerUrlMapping,
+  TriggerConfig,
+  TRIGGER_CALLER,
+  TriggerMapping,
 } from './trigger-config';
 import { OutletService } from '../../../../cms-structure/outlet/index';
 
 @Injectable({ providedIn: 'root' })
 export class TriggerService {
-  constructor(protected outletService: OutletService<ComponentFactory<any>>) {}
+  // Keep a list of rendered elements
+  protected renderedCallers: TRIGGER_CALLER[] = [];
 
-  renderDialog(
-    config: TriggerOutletMapping | TriggerInlineMapping | TriggerUrlMapping,
-    template?: ComponentFactory<any>
-  ): void | string {
+  constructor(
+    protected outletService: OutletService<ComponentFactory<any>>,
+    protected triggerConfig: TriggerConfig,
+    protected componentFactoryResolver: ComponentFactoryResolver
+  ) {}
+
+  renderDialog(caller: TRIGGER_CALLER, component?: any): void | string {
+    const config = this.findConfiguration(caller);
     if (Boolean((config as TriggerUrlMapping).url)) {
+      this.renderedCallers.push(caller);
       return (config as TriggerUrlMapping).url;
-    } else if (Boolean((config as TriggerOutletMapping).outlet)) {
-      this.renderOutlet(config as TriggerOutletMapping, template);
-    } else if (Boolean((config as TriggerInlineMapping).inline)) {
-      this.renderInline(config as TriggerInlineMapping, template);
+    } else if (
+      this.shouldRender(caller, config as TriggerMapping) &&
+      component
+    ) {
+      const template = this.componentFactoryResolver.resolveComponentFactory(
+        component
+      );
+      if (Boolean((config as TriggerOutletMapping).outlet)) {
+        this.renderOutlet(config as TriggerOutletMapping, template);
+      } else if (Boolean((config as TriggerInlineMapping).inline)) {
+        this.renderInline(config as TriggerInlineMapping, template);
```

Comment:
My idea about the inline rendering was this:
- take the trigger element
- render the element next to the trigger (BEFORE / AFTER could stil work there)

I don't think we should render it as an outlet to the cx-storefront.

Reply:
Makes sense, I wanted to clarify that with you.